theme/static/js/hs_resource_table.js

```diff
@@ -327,8 +340,13 @@ function delete_multiple_resources_ajax_submit(indexes) {
                 url: url,
                 datastring: datastring,
                 dataType: "html",
-                success: function () {
-                    resourceTable.row(row).remove();  // Delete row from the table
+                success: function (task) {
+                    task = JSON.parse(task)
+                    notificationsApp.registerTask(task);
+                    notificationsApp.show();
+                    setTimeout(function() {
+                            check_delete_resource_task(task, row);
```

Comment:
I think the notificationsApp should just check each task in the list on it's own.  Code outside of the notificationsApp should really only need to call registerTask.

Reply:
@sblack-usu That would inflate notificationsApp greatly as more and more tasks are being added. In addition, in this special case of the same resource deletion task being invoked from both single resource deletion and multiple resource deletion from different pages, notificationsApp would need to know how to handle them differently if all the logic is moved to the notificatoinsApp. I believe the way it is implemented currently is better in terms of the flexibility and extensibility with a good balance between having notificationsApp to handle common task management notification and leaving the details of different processing for each task outside the notificationsApp.

Reply:
I don't think the notificationsApp should do anything on completion outside of a notification.  With the delete resource task, it doesn't make sense to force a redirect after the task completes.  I think we should just notify the user.  Because we're now running these tasks asynchronously, we need to rethink the workflows.

If we simply notified on task completion, we could generalize for all tasks too.

Reply:
I believe that was what I was initially aimed at. I added a property in the notification app to notify the caller the task status. However, the caller in this case is an ajax call, and I don't know how the notification app can proactively notify the caller without the caller polling the notification app for task status. If the notification app raises a signal, how can we set up for the caller to listen to this signal? I am not proficient with JS enough to know how to achieve this at this point. It seems to me that some JS framework needs to be brought in for this. Since @cuahsimarko already brought in more JS framework libs for his discover work, my plan is to get this implemented first with reasonable extensibility and maintainability for now, then when @cuahsimarko starts to bring in more JS framework into the whole HS stack, this workflow could be improved further at that time. However, if you have ideas and know how to improve this further at this time, that would be great.

Reply:
This is already using vue, which was introduced by @cuahsimarko.  I think the vue notificationApp should just poll all known tasks on it's own.  That way we can register a task and then forget about it, letting the notificationApp handle things from there.

Reply:
Yes, that is what the vue notificationApp is currently doing. The problem is how to connect between the caller, e.g., multiple resource deletion ajax request from my resources page that registered the task, and the notification app when the task is done if we don't use timer in the caller to poll the notification app to know the task status. In other words, the caller can register the task to let notification app do the job, but after the task completes, how the notification app notifies the caller, in this case, an ajax call from my resources page, when the task completes? If the caller does not use the time to poll the task status from the notification app, the ajax call will complete immediately and there is no way for the notification app to get back to the original ajax call to do the UI updates as needed after the task completes.

Reply:
I don't think we should automatically do the UI updates anymore, maybe just notify it's complete with a suggestion to refresh or something... We don't even know the user will still be on the resource landing page.  A UI update doesn't make sense in that case.  And it would be a nuisance to the user if they're elsewhere in the system and get redirected back to the resource because an async task completes.

Maybe @cuahsimarko can chime in, there may be things that vue can do we are not aware of.  We really do need a frontend developer to think through some of these problems more thoroughly.  Unfortunately, Mauriel isn't available much anymore, so I think we'll need @cuahsimarko to join this review.

Reply:
That is an alternative way to handle it than the UI automatic update. It is true the user may be away in which case the task trigger ajax call for checking task completion will be bailed out anyway, and when user comes back after the task completes, the page will be refreshed with updated UI. I don't think they will be redirected back to the resource because an async task completes as currently implemented, but if you have tested it out and it did happen, then I agree the alternative way would work better for this specific case. However, there are problems even with the alternative approach. For example, when the user clicks on the refresh link/button after the task completes, what it will trigger or leads to? It sends another refresh request for the server to respond to? I don't think the alternative workflow would be more streamlined than the current implementation in terms of code implementation unless I misunderstood the workflow or missed something.